main.tf

```diff
@@ -70,6 +70,76 @@ resource "aws_placement_group" "placement_group" {
   depends_on = [module.network]
 }
 
+resource "aws_kms_key" "kms_key" {
+  count                   = var.ebs_encrypted && var.kms_key_arn == "" ? 1 : 0
```

Comment:
lets name is `kms_key_id` instead of `kms_key_arn` and let have local var for this `if` `var.ebs_encrypted && var.kms_key_arn == ""` we can name it `create_kms_key`

Reply:
actually to make it more clear, lets call is `ebs_kms_key_id`
In addition we need to have it on clients and protocol gws as well